rocksdb_admin/admin_handler.cpp

```diff
@@ -1095,18 +1136,55 @@ void AdminHandler::async_tm_startMessageIngestion(
       common::Stats::get()->AddMetric(kKafkaConsumerLatency, latency_ms);
     }
 
-    // Write the message to rocksdb
-    static const rocksdb::WriteOptions write_options;
     auto key = rocksdb::Slice(static_cast<const char *>(message->key_pointer()),
                               message->key_len());
-    auto val = rocksdb::Slice(static_cast<const char *>(message->payload()),
-                              message->len());
-    auto status = db->rocksdb()->Put(write_options, key, val);
-    if (!status.ok()) {
-      // TODO: if there are transient errors, add a retry
-      LOG(ERROR) << "Failure while writing to " << db_name << ": "
-                 << status.ToString();
-      common::Stats::get()->Incr(kKafkaDbWriteErrors);
+
+    // Deserialize the kafka payload
+    KafkaOperationCode op_code;
+    rocksdb::Slice value;
+    if (should_deserialize) {
+      std::string deser_val;
```

Comment:
move it out side if() {}, otherwise value will be reference to invalid content when outside the if block.